java-checks/src/main/java/org/sonar/java/checks/BigDecimalDoubleConstructorCheck.java

```diff
@@ -56,16 +59,34 @@ public void visitNode(Tree tree) {
         .forRule(this)
         .onTree(tree)
         .withMessage("Use \"BigDecimal.valueOf\" instead.");
-      if (newClassTree.arguments().size() == 1) {
-        builder.withQuickFix(() -> quickFix(newClassTree));
+
+      Arguments arguments = newClassTree.arguments();
+      if (arguments.size() == 1) {
+        builder.withQuickFix(() -> valueOfQuickFix(newClassTree));
+      } else {
+        ExpressionTree firstArgument = arguments.get(0);
+        if (firstArgument instanceof LiteralTree) {
+          builder.withQuickFix(() -> stringConstructorQuickFix((LiteralTree) firstArgument));
+        }
       }
       builder.report();
     }
   }
 
-  private static JavaQuickFix quickFix(NewClassTree newClassTree) {
+  private static JavaQuickFix valueOfQuickFix(NewClassTree newClassTree) {
     return JavaQuickFix.newQuickFix("Replace with BigDecimal.valueOf")
       .addTextEdit(JavaTextEdit.replaceBetweenTree(newClassTree.newKeyword(), newClassTree.identifier(), "BigDecimal.valueOf"))
       .build();
   }
+
+  private static JavaQuickFix stringConstructorQuickFix(LiteralTree argument) {
+    String argumentValue = argument.value();
+    if (argumentValue.endsWith("f") || argumentValue.endsWith("d")) {
```

Comment:
Warning, the decimal suffix is case insensitive: `1.0f` == `1.0F` and `1.0d` == `1.0D`